### ISSUE TITLE
Package/program name conflict

### ISSUE THREAD

**Issue by hwti** [natesales/q#28]
On at least Fedora and Archlinux, there are already packages with the same name and executable name.

https://packages.fedoraproject.org/pkgs/q/q/
https://aur.archlinux.org/packages/q

This means that if I install `q_0.8.2_linux_amd64.rpm` on Fedora, `dnf` see the distribution `q` package as an update.

**Comment by seththeriault** [comment]
This also conflicts with an existing Homebrew package:

https://formulae.brew.sh/formula/q

**Comment by natesales** [comment]
The homebrew package can be installed with `brew install natesales/repo/q` after adding the natesales tap.

**Comment by holgersson32644** [comment]
First of all, thanks for that nifty CLI tool ;)

FTR: In Gentoo/Linux we have a similiar problem, as the name "q" is already taken by [a package related to package management](https://packages.gentoo.org/packages/app-portage/portage-utils).
In my overlay (some kind of an unofficial repository) I packaged `q` as `net-misc/q` but renamed the binary [into `q-dns` there](https://git.holgersson.xyz/foss/holgersson-overlay/src/branch/master/net-misc/q/q-0.8.3.ebuild#L52).

Are you interested in some "public brainstorming" to find a short but distinct name for q?
I propose one of the following
- `q-dns` (or `qdns`)
- `q-ns` (or `qns`)
- `magical-creature-from-startrek` ;)

WDYT?

**Comment by Seirdy** [comment]
I strongly recommend changing the binary name to something unique, as a name conflict will prevent creating distribution packages. `qdns` is taken by another smaller DNS-related project, unfortunately; however, `q-dns` seems safe. I prefer `qdig` as it's less ambiguous and easier to remember (it's like `dig`, but with DoQ!)

There's also `obdig` (`dig` with <b>OB</b>livious DNS over HTTPS), which highlights its unique ODOH capabilities.

Checking Repology, the AUR, GitHub, and Fedora package+binary name conflicts: `qdig` and `obdig` seem fine.

**Comment by classabbyamp** [comment]
any updates on this? I'm working on packaging this for void linux, and we generally keep with the upstream naming, but if the name is going to change I'd prefer not needing to change the package's name

**Comment by rahilarious** [comment]
This is a really handy tool. 

Besides problems in packaging, project @natesales  is hurting itself by not naming it in search-friendly name. Because of this single letter name, there is no branding as well. How would somebody quickly tell your friend about this utility? "hey, checkout this `q`?" One has to either send link or remember your name and tell "checkout `natesales/q`" which is highly unlikely to happen. 

Author is not living upto his last name "Sales" cause this is not great salesmanship :-)

**Comment by polarathene** [comment]
`q-dns` or `qig` both seem fine 👍 (_`qdig` / `q-dig` are a bit easier to pronounce though and more familiar `dig` association to DNS CLI tools_)

Agree that it's not the easiest name to discover, I just search "q dns github" if I'm lazy 😅 

`doggo` is a competing option, if you want to stick with the idea of similar names to others (`qdns`) you could go with `doqqo` (_1 less letter, efficient!_ 😂 ). Both dig/doggo based variations technically blend in with all the others in the [README comparison table](https://github.com/natesales/q#feature-comparison). `goq` is used elsewhere, but `doqqo` is fairly unique if you don't mind the overlap with `doggo` in search results (_probably a bonus, besides some potential user confusion_).

I'm not familiar with packaging, would it be much of an issue if another package installs a binary of the same name too? Or is this only about the package name? Users could always alias a longer name to `q` if they like 🤷‍♂️

**Comment by den-is** [comment]
I really like `q` as alternative to everything, especially as reincarnation of dead `dog`

I've installed `q` with brew. Looks like I will have to re-install it manually from binary release with a different longer name.

`q` name choice was too bold, cocky, brave, audacious, and impudent for such a small but still public project.

Here is another clash `Amazon Q`:
https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-getting-started-installing.html

**Comment by polarathene** [comment]
> I really like `q` as alternative to everything, especially as reincarnation of dead `dog`

Was "`q` as an alternative" a typo?

Did you mean `doq` / `doqqo`? (_given your reference to `dog` and the 👍 to my prior comment_)

**Comment by den-is** [comment]
> > I really like `q` as alternative to everything, especially as reincarnation of dead `dog`
> 
> Was "`q` as an alternative" a typo?
> 
> Did you mean `doq` / `doqqo`? (_given your reference to `dog` and the 👍 to my prior comment_)

I meant that before the `q` I was using [dog](https://github.com/ogham/dog) the utility which became deprecated.
I didn't know about the `doggo`.
IMHO `doq` is a "so-so" option as it sounds like "doc (document)"

**Comment by polarathene** [comment]
> IMHO `doq` is a "so-so" option as it sounds like "doc (document)"

Agreed, which is why I didn't suggest it. I was just trying to make sense of what your line meant, thinking it was about the naming topic 😂 (_I understand now that it was expressing your preference as a alternative DNS utility to `dog`, nothing name related_)

I figured `doqqo` was a good fit, keeping the `q` origin, while using it with the more familiar `doggo` which also happens to be Go (_**EDIT:** Both projects have approx 1.5k stars, but I have seen `doggo` mentioned  in communities more often_).

---

> I didn't know about the `doggo`.

I'm not sure how it compares to `q` these days (_I don't think the comparison table on `q` README has been revised for a while?_).

**Comment by den-is** [comment]
Another clash https://github.com/harelba/q (well it is the same pretense to the other tool's author)

**Comment by lapo-luchini** [comment]
I checked the build files and I don't understand how this installs as `q-dns` in FreeBSD and  `q` in Arch. 🤣
My muscle memory would really like a common name everywhere.

**Comment by k0ste** [comment]
> This also conflicts with an existing Homebrew package:
> 
> https://formulae.brew.sh/formula/q

Now Formula control under q (this) package, since https://github.com/Homebrew/homebrew-core/pull/250581

**Comment by Gunni** [comment]
I always have to re-find this project when i notice i'm missing it on a new system, it takes several googles to find it every time... It needs a findable/searchable/memorable name, even if the cli command is q...

**Comment by polarathene** [comment]
> It needs a findable/searchable/memorable name, even if the cli command is q...

`natesales/q` 🤓

`q-dns` is often what the project is packaged under elsewhere and if you use that as a search term the project repo here comes up as the first result on Google for me, all you need to remember is to append `-dns` to `q`? 🤷‍♂️
